### PR TITLE
Add videofile processing

### PR DESCRIPTION
**Pull request by lwp-ygavrilenko**
Resolve issue (#21)

### CODE REVIEW

File: app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java [hunk 1]
```diff
@@ -323,21 +331,91 @@ public Long addNewDeck(String deckName) {
 
     // @todo: refactor once new version release of "com.ichi2.anki.api" is available
     public String addFileToAnkiMedia(String uriString) {
+        Uri uri = Uri.parse(uriString);
```
Comment (lwp-emelnik) [inline app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java:334]: Not really sure that this extractor should be placed inside AnkiDroidHelper

File: app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java [hunk 2]
```diff
@@ -323,21 +331,91 @@ public Long addNewDeck(String deckName) {
...
+        mContext.grantUriPermission("com.ichi2.anki", uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
```
Comment (lwp-emelnik) [inline app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java:335]: I think there should be a special constant for "com.ichi2.anki" (package name or something like that)
if not, please create/move it somewhere

File: app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java [hunk 3]
```diff
@@ -323,21 +331,91 @@ public Long addNewDeck(String deckName) {
...
+
+                if (mediaMuxer == null) {
+                    mContext.revokeUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    return null;
+                }
+
+                MediaCodec.BufferInfo bufferInfo = new MediaCodec.BufferInfo();
+                bufferInfo.presentationTimeUs = 0;
+
+                ByteBuffer byteBuffer = ByteBuffer.allocate(500 * 1024);
```
Comment (lwp-emelnik) [inline app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java:374]: Please use some contants for "magic numbers" or describe in comments what is that

File: app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java (hunk 1 above)
Comment (lwp-ygavrilenko) [reply app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java:329]: Good point. Factored audio extraction into utility class

File: app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java (hunk 3 above)
Comment (lwp-ygavrilenko) [reply app/src/main/java/com/ichi2/apisample/helper/AnkiDroidHelper.java]: Fixed

**Pull request merged**